docs/testing.md

```diff
@@ -0,0 +1,71 @@
+---
+id: testing
+title: Testing Fuse
+---
+
+We use `sbt`'s built-in testing facilities to run tests. There are four test
+suites for the language:
+
+## Parsing Tests
+
+Parsing tests simply check to see if language constructs are parsable or not.
+They do not perform any type checking
+
+## Type Checker Spec
+
+These is the most comprehensive set of example to document the behavior of
+all language constructs with respect to the type checker. We use the
+[FunSpec](http://www.scalatest.org/getting_started_with_fun_spec) model to
+specify tests:
+
+- Create a `describe` for the new language construct. If there are too many
+  tests for one construct, consider splitting them using logical groups. For
+  example, "capabilities w/ simple contexts" and "capabilities w/ complex contexts"
+
+- In each describe, use the `it` function to give the test a descriptive name
+  and use the `typeCheck` method to write the test. Use Scala's [multiline
+  strings](https://www.oreilly.com/library/view/scala-cookbook/9781449340292/ch01s03.html)
+  to specify long programs.
+
+## File tests
+
+The file test harness simply pulls in tests under `src/test/{should-compile,should-fail}`.
+As the names suggest, the first directory contains tests that should compile with
+the type checker while the second contains programs that should fail.
+
+Simply add a Fuse program under the directory to get them to run.
+
+## Run tests
+
+This harness is used to check the functional correctness of the emitted code.
+It creates a temporary directory `_runTestTmp` to store intermediary C++ files.
```

Comment:
The only benefit is that when tests fail, you can look at the C++ generated under that deterministically generated directory. Thoughts?

Reply:
Got it. Both options seem reasonable, (cd'ing to /tmp would be frustrating), but since the goal is convenience, I recommend calling it something simple like `_test` or `_build` or `_tmp`.

Reply:
Renamed to `_test`.